Tasks/ServiceFabricComposeDeployV0/ServiceFabricComposeDeploy.ps1

```diff
@@ -9,271 +9,286 @@ Import-Module $PSScriptRoot\ps_modules\TelemetryHelper
+            Write-Host (Get-VstsLocString -Key CreatingApplication)
+            New-ServiceFabricComposeApplicationHelper -ApiVersion $apiVersion -DeployParameters $deployParameters
 
-        Write-Host (Get-VstsLocString -Key WaitingForDeploy)
-        $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
-        while (($newApplication -eq $null) -or `
-            ($newApplication.Status -eq 'Provisioning') -or `
-            ($newApplication.Status -eq 'Creating'))
-        {
-            if ($newApplication -eq $null)
+            Write-Host (Get-VstsLocString -Key WaitingForDeploy)
+            $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
+            while (($newApplication -eq $null) -or `
+                ($newApplication.Status -eq 'Provisioning') -or `
+                ($newApplication.Status -eq 'Creating'))
             {
-                Write-Host (Get-VstsLocString -Key WaitingForDeploy)
+                if ($newApplication -eq $null)
+                {
+                    Write-Host (Get-VstsLocString -Key WaitingForDeploy)
+                }
+                else
+                {
+                    Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $newApplication.Status)
+                }
+                Start-Sleep -Seconds 3
+                $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
             }
-            else
+            Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $newApplication.Status)
+
+            if ($newApplication.Status -ne 'Created' -and $newApplication.Status -ne 'Ready')
             {
-                Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $newApplication.Status)
+                Write-Error (Get-VstsLocString -Key DeployFailed -ArgumentList @($newApplication.Status.ToString(), $newApplication.StatusDetails))
             }
-            Start-Sleep -Seconds 3
-            $newApplication = Get-ServiceFabricComposeApplicationStatusHelper -ApiVersion $apiVersion -GetStatusParameters $getStatusParameters
         }
-        Write-Host (Get-VstsLocString -Key CurrentStatus -ArgumentList $newApplication.Status)
-
-        if ($newApplication.Status -ne 'Created' -and $newApplication.Status -ne 'Ready')
+    }
+    finally
+    {
+        if ($null -ne $certificate)
         {
-            Write-Error (Get-VstsLocString -Key DeployFailed -ArgumentList @($newApplication.Status.ToString(), $newApplication.StatusDetails))
+            $thumbprint = $certificate.Thumbprint
+            $certificate | Get-Member
+            if (Test-Path "Cert:\CurrentUser\My\$thumbprint")
+            {
+                Remove-Item "Cert:\CurrentUser\My\$thumbprint" -Force
```

Comment:
Any error in removing cert should be suppressed

Reply:
I was debating outright suppressing them, which is why I used two finally blocks (didn't want any error to remove telemetry) will add suppression and remove extra finally.